src/Admin/SiteHealth.php

```diff
@@ -20,18 +24,50 @@
  *
  * @since 1.5.0
  */
-final class SiteHealth {
+final class SiteHealth implements Service, Registerable, Delayed, Conditional {
+
+	/**
+	 * Service that monitors and controls the CSS transient caching.
+	 *
+	 * @var MonitorCssTransientCaching
+	 */
+	private $css_transient_caching;
+
+	/**
+	 * Check whether the conditional object is currently needed.
+	 *
+	 * @return bool Whether the conditional object is needed.
+	 */
+	public static function is_needed() {
+		return is_admin() && ! wp_doing_ajax();
+	}
+
+	/**
+	 * Get the action to use for registering the service.
+	 *
+	 * @return string Registration action to use.
+	 */
+	public static function get_registration_action() {
+		return 'wp_loaded';
+	}
```

Comment:
This demonstrates well how, with this new architecture, all the relevant decison-making for a given subsystem/service is in one location. No more chasing around across files and doing changes in multiple places to satisfy a single requirements change.